src/Symfony/Component/Validator/Constraints/ExpressionLanguageSyntax.php

```diff
@@ -29,8 +29,7 @@ class ExpressionLanguageSyntax extends Constraint
 
     public $message = 'This value should be a valid expression.';
     public $service;
-    public $validateNames = true;
-    public $names = [];
+    public $variableNames = [];
```

Comment:
`$allowedVariables` would be better to me (as suggested in the linked doc issue btw)

Reply:
oh, and shouldn't the default be `null`, to allow any variables?

Reply:
Indeed, the suggested name sounds better to me. I am going to change that.

My reasoning for the empty array as the default value was that allowing arbitrary names rather looks like an edge case to me.

Reply:
Not allowing any variables by default means allowing only static expressions, isn't it? Since static expressions are never why ppl use EL, this would effectively force ppl to be explicit all the time.

Reply:
I applied the suggested changes while merging, please tell if you think this is a mistake :)

Reply:
Works for me. :) Thanks!